libusb/os/linux_usbfs.c

```diff
@@ -204,16 +204,19 @@ static int get_usbfs_fd(struct libusb_device *dev, int access_mode, int silent)
 	if (errno == ENOENT) {
 		const long delay_ms = 10L;
 		const struct timespec delay_ts = { 0L, delay_ms * 1000L * 1000L };
+		uint8_t retry = 3;
 
-		if (!silent)
-			usbi_err(ctx, "File doesn't exist, wait %ld ms and try again", delay_ms);
+		while (retry-- > 0) {
+			if (!silent)
+				usbi_err(ctx, "File doesn't exist, wait %ld ms and try again", delay_ms);
```

Comment:
Could you please log errno (ideally store it on error and log it from that variable). "File doesn't exist" might be correct, but it could also be a permission error and I think it would be good if one is able to tell the difference.

Reply:
Ah, I realise that this *is* ENOENT in the first loop iteration.

In that case, maybe the loop should break if errno is != ENOENT?